test/integ/test_key_pair_authentication.py

```diff
@@ -14,6 +17,46 @@
 import snowflake.connector
 
 
+@pytest.mark.skipolddriver
+def test_get_token_from_private_key():
+    test_user = "python_test_keypair_user_" + str(uuid.uuid4()).replace("-", "_")
+    current_dir = path.dirname(path.realpath(__file__))
+    private_key_file_path = path.join(
+        current_dir, "../data/rsa_keys", "rsa_key_encrypted.p8"
+    )
+    private_key_password = "test"
+    public_key_fingerprint = snowflake.connector.auth.get_public_key_fingerprint(
+        private_key_file_path, private_key_password
+    )
+    # Create a dict containing input accounts and their corrected account fields used to generate the correct token
+    accounts = {
+        "s3testaccount.global": "S3TESTACCOUNT.GLOBAL",
+        "acct-with-dashes": "ACCT_WITH_DASHES",
+        "testaccount.extra": "TESTACCOUNT",
+    }
+    for a in accounts:
```

Comment:
You should use [parametrized tests](https://docs.pytest.org/en/stable/parametrize.html). We use them all over the code-base.